examples/plot_knn_s2.py

```diff
@@ -0,0 +1,68 @@
+"""Plot the result of a KNN classification on the sphere."""
+
+import logging
+import os
+
+import matplotlib.pyplot as plt
+
+import geomstats.backend as gs
+import geomstats.visualization as visualization
+from geomstats.geometry.hypersphere import Hypersphere
+from geomstats.learning.knn import KNearestNeighborsClassifier
+
+
+def main():
+    """Plot the result of a KNN classification on the sphere."""
+    sphere = Hypersphere(dimension=2)
+    sphere_distance = sphere.metric.dist
+
+    n_labels = 2
+    n_samples_per_dataset = 10
+    n_targets = 200
+
+    dataset_1 = sphere.random_von_mises_fisher(
+        kappa=10,
+        n_samples=n_samples_per_dataset)
+    dataset_2 = - sphere.random_von_mises_fisher(
+        kappa=10,
+        n_samples=n_samples_per_dataset)
+    training_dataset = gs.concatenate((dataset_1, dataset_2), axis=0)
+    labels_dataset_1 = gs.zeros([n_samples_per_dataset], dtype=int)
+    labels_dataset_2 = gs.ones([n_samples_per_dataset], dtype=int)
+    labels = gs.concatenate((labels_dataset_1, labels_dataset_2))
+    target = sphere.random_uniform(n_samples=n_targets)
+
+    neigh = KNearestNeighborsClassifier(n_neighbors=2,
```

Comment:
NIT: for indentation, I usually prefer the syntax:
```
neigh = KNearestNeighborsClassifier(
    n_neighbors=2, metric=sphere_distance)
```

because if you change the variable name `neigh` to a name with a different number of characters, you have to re-align all your code, whereas the above syntax does not need it!